net-vpn/i2p/i2p-0.9.48.ebuild

```diff
@@ -0,0 +1,159 @@
+		dev-java/ant-junit4:0
+		dev-java/hamcrest-core:1.3
+		dev-java/hamcrest-library:1.3
+		dev-java/junit:4
+	)
+"
+
+RDEPEND="${CP_DEPEND}
+	acct-user/i2p
+	acct-group/i2p
+	virtual/jre:1.8
+	net-libs/nativebiginteger:0
+"
+
+EANT_BUILD_TARGET="pkg"
+# no scala as depending on antlib.xml not installed by dev-lang/scala
+EANT_TEST_TARGET="junit.test"
+JAVA_ANT_ENCODING="UTF-8"
+
+src_prepare() {
+	if use test; then
+		# no *streaming as requiring >dev-java/mockito-1.9.5
+		sed -e "/streaming.*junit\.test/d" \
+			-i build.xml ||
+			die "unable to remove ministreaming tests"
+	fi
+
+	# as early as possible to allow generic patches to be applied
+	default
+
+	java-ant_rewrite-classpath
+
+	java-pkg-2_src_prepare
+
+	# We're on GNU/Linux, we don't need .exe files
+	echo "noExe=true" > override.properties || die
+	if ! use nls; then
+		echo "require.gettext=false" >> override.properties || die
+	fi
+
+	# avoid auto starting browser
+	sed -i "s|clientApp.4.startOnLoad=true|clientApp.4.startOnLoad=false|" \
+		"installer/resources/clients.config" || die
+
+	# generate wrapper classpath, keeping the default to be replaced later
+	i2p_cp="" # global forced by java-pkg_gen-cp
+	java-pkg_gen-cp i2p_cp
+	local lib i=2
+	local cp="wrapper.java.classpath.1=${EPREFIX}/usr/share/i2p/lib/*\n"
+	for lib in ${i2p_cp//,/ }
+	do
+		cp+="wrapper.java.classpath.$((i++))=$(java-pkg_getjars ${lib})\n"
+	done
+
+	# add generated cp, hardcode system VM, setting system's conf
+	sed -e "s|\(wrapper\.java\.classpath\.1\)=.*|${cp}|" \
+		-e "s|\(wrapper\.java\.command\)=.*|\1=/etc/java-config-2/current-system-vm/bin/java|" \
+		-e "s|\(wrapper\.java\.library\.path\.1\)=.*|\1=/usr/$(get_libdir)/java-service-wrapper|" \
+		-e "s|\(wrapper\.java\.library\.path\)\.2=.*|\1.2=${EPREFIX}/usr/share/i2p/lib\n\1.3=/usr/$(get_libdir)|" \
+		-e "s|\(wrapper\.java\.additional\.1=-DloggerFilenameOverride\)=.*|\1=${EPREFIX}/var/log/i2p/router-@|" \
+		-e "s|\(wrapper\.logfile\)=.*|\1=${EPREFIX}/var/log/i2p/wrapper|" \
+		-e "/wrapper\.java\.additional\.2\(\.stripquote\|\)/d" \
+		-i installer/resources/wrapper.config ||
+		die "unable to apply gentoo config"
+	local prop i=2
+	for prop in \
+		"i2p.dir.base=${EPREFIX}/usr/share/i2p" \
+		"i2p.dir.app=${EPREFIX}/var/lib/i2p/app" \
+		"i2p.dir.config=${EPREFIX}/var/lib/i2p/config" \
+		"i2p.dir.router=${EPREFIX}/var/lib/i2p/router" \
+		"i2p.dir.log=${EPREFIX}/var/log/i2p" \
+		"i2p.dir.pid=${EPREFIX}/tmp" \
+		"i2p.dir.temp=${EPREFIX}/tmp"
+	do
+		echo "wrapper.java.additional.$((i++))=-D$prop" >> installer/resources/wrapper.config ||
+			die "unable to apply gentoo config"
+	done
+}
+
+src_test() {
+	# store built version of jars, overwritten by testing
+	mv "${S}/pkg-temp/lib/"{i2p,router}.jar "${T}" ||
+		die "unable to save jars before tests"
+
+	# generate test classpath
+	local cp
+	cp="$(java-pkg_getjars --build-only junit-4,hamcrest-core-1.3,hamcrest-library-1.3)"
+	EANT_TEST_EXTRA_ARGS="-Djavac.classpath=${cp}" java-pkg-2_src_test
```

Comment:
Could you rename this variable as it immediately made me think of the Unix `cp` command which in return churned my brain to a know for a moment.

Reply:
Will do in next bump then.